app/pages/project/vpcs/RouterRoutePage.tsx

```diff
@@ -0,0 +1,212 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import { createColumnHelper } from '@tanstack/react-table'
+import { useCallback, useMemo } from 'react'
+import { Outlet, useNavigate, type LoaderFunctionArgs } from 'react-router-dom'
+
+import { Networking16Icon, Networking24Icon } from '@oxide/design-system/icons/react'
+
+import {
+  apiQueryClient,
+  useApiMutation,
+  usePrefetchedApiQuery,
+  type RouterRoute,
+} from '~/api'
+import { DocsPopover } from '~/components/DocsPopover'
+import { HL } from '~/components/HL'
+import { MoreActionsMenu } from '~/components/MoreActionsMenu'
+import { routeFormMessage } from '~/forms/vpc-router-route/shared'
+import { getVpcRouterSelector, useVpcRouterSelector } from '~/hooks'
+import { confirmAction } from '~/stores/confirm-action'
+import { addToast } from '~/stores/toast'
+import { EmptyCell } from '~/table/cells/EmptyCell'
+import { TypeValueCell } from '~/table/cells/TypeValueCell'
+import { useColsWithActions, type MenuAction } from '~/table/columns/action-col'
+import { PAGE_SIZE, useQueryTable } from '~/table/QueryTable'
+import { Badge } from '~/ui/lib/Badge'
+import { CreateLink } from '~/ui/lib/CreateButton'
+import { DateTime } from '~/ui/lib/DateTime'
+import { EmptyMessage } from '~/ui/lib/EmptyMessage'
+import { PageHeader, PageTitle } from '~/ui/lib/PageHeader'
+import { PropertiesTable } from '~/ui/lib/PropertiesTable'
+import { TableControls, TableTitle } from '~/ui/lib/Table'
+import { docLinks } from '~/util/links'
+import { pb } from '~/util/path-builder'
+
+RouterRoutePage.loader = async function ({ params }: LoaderFunctionArgs) {
+  const { project, vpc, router } = getVpcRouterSelector(params)
+  await Promise.all([
+    apiQueryClient.prefetchQuery('vpcRouterView', {
+      path: { router },
+      query: { project, vpc },
+    }),
+    apiQueryClient.prefetchQuery('vpcRouterRouteList', {
+      query: { project, router, vpc, limit: PAGE_SIZE },
+    }),
+  ])
+  return null
+}
+
+const RouterRouteTypeValueBadge = ({ type, value }: { type: string; value?: string }) => {
+  const typeString = type
+    .replace('_', ' ')
+    .replace('ip net', 'ip network')
+    .replace('internet gateway', 'gateway')
+    .replace('subnet', 'VPC subnet')
```

Comment:
This should be done as a record like `destTypes` and `targetTypes` to make sure we're always catching all possible cases. Slightly awkward question of whether we should use exactly the same abbreviations in both spots. I am leaning no, since "Internet gateway" looks ridiculous in the badge but it probably looks fine in the listbox on the form. So I guess I'd say make a separate mapping in this file for this purpose.